app/manage/pods/device-details/share-modal-directive/index.js

```diff
@@ -12,18 +12,42 @@ module.exports = [
         '$scope',
         '$timeout',
         '$location',
-        function($scope, $timeout, $location) {
+        'Sms',
+        function($scope, $timeout, $location, Sms) {
 
           // template variables -------------------
           $scope.copySuccess = false;
           $scope.url = $location.absUrl().replace('?demo=1', '');
+          $scope.phone = null;
 
           // create clipboard ---------------------
           var clipboard = new Clipboard('.copy-btn');
           clipboard.on('success', function(e) {
-            $scope.copySuccess = true;
+            $scope.messageText = 'Copied!';
           });
 
+          // send text ----------------------------
+          $scope.sendText = function() {
+            var data = {
+              'phone': $scope.phone,
+              'message': $scope.url
+            };
+
+            if (!$scope.phone) {
+              $scope.messageText = 'Please enter a number!';
```

Comment:
I'm excited too but not sure they should be scolded with an exclamation point!! :p